kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,275 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafkarest;
+
+import io.confluent.kafkarest.entities.ConsumerGroup;
+import io.confluent.kafkarest.entities.ConsumerGroupCoordinator;
+import io.confluent.kafkarest.entities.ConsumerGroupSubscription;
+import io.confluent.kafkarest.entities.ConsumerTopicPartitionDescription;
+import io.confluent.kafkarest.entities.Topic;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.ConsumerGroupListing;
+import org.apache.kafka.clients.admin.MemberDescription;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.JavaConverters;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.Properties;
+import java.util.HashSet;
+import java.util.Comparator;
+
+
+public class GroupMetadataObserver {
+
+  private static KafkaConsumer<Object, Object> createConsumer(String groupId,
+                                                              KafkaRestConfig appConfig) {
+    final Properties properties = new Properties();
+    String deserializer = StringDeserializer.class.getName();
+    properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+            RestConfigUtils.bootstrapBrokers(appConfig));
+    properties.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+    properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "30000");
+    properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, deserializer);
+    properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, deserializer);
+    properties.putAll(appConfig.getConsumerProperties());
+
+    return new KafkaConsumer<>(properties);
+  }
+
+  private final Logger log = LoggerFactory.getLogger(GroupMetadataObserver.class);
+
+  private final KafkaRestConfig config;
+  private final AdminClientWrapper adminClientWrapper;
+
+  public GroupMetadataObserver(KafkaRestConfig config, AdminClientWrapper adminClientWrapper) {
+    this.config = config;
+    this.adminClientWrapper = adminClientWrapper;
+  }
+
+  /**
+   * <p>Get consumer group list</p>
+   *
+   * @return list of consumer groups
+   */
+  public List<ConsumerGroup> getConsumerGroupList(Option<Integer> offsetOpt,
```

Comment:
Here and everywhere else: Why are you using Scala's `Option` instead of Java's `Optional`? Please, use `Optional`. Also, avoid methods with `Optional` parameters. The idiomatic way of doing that in Java is with method overloads, e.g.:

```
void foo(); // without parameter
void foo(int bar); // with parameter
```